docs: tutorial: Upload photos using the Open Food Facts API (#7871)
openfoodfacts/openfoodfacts-server: docs/tutorials/uploading-photo-to-a-product.md

> # Uploading Images to the OFF API

Reviewer: ```suggestion
# Uploading Images to the Open Food Facts API
```

<!--Add a brief introduction of what the tutorial does -->

> This basic tutorial shows you can upload image of a product to the OFF API.

Reviewer: ```suggestion
This basic tutorial shows you can upload image of a product to the Open Food Facts API.
```

### Upload Behavior

In case you upload more than one photo of the front, the ingredients and the nutrition facts, beware that only the first photo of each category will be displayed. (You might want to take additional images of labels, recycling instructions, and so on). All photos will be saved.

### Label Languages

Multilingual products have several photos based on languages present on the packaging. You can specify the language by adding a lang code suffix to the request.

## Authentication

The WRITE operations in the OFF API require authentication, therefore you need a valid `user_id` and `password`  to write the photo to 100% Real Orange Juice.

> Sign up on the [Open Food Facts App](https://world.openfoodfacts.net/), to get your `user_id` and `password` if you dont have.
For more details, visit the [Open Food Facts Authentication](https://openfoodfacts.github.io/openfoodfacts-server/introduction/api/#authentication).

## Parameters

### Code

The barcode of the product.

### Imagefield

> Imagefield indicates the type of the image you are trying to upload for a particular product. It can be either of these: `front`, `ingredients`, `nutrition`, `packaging` or `other`.

Reviewer: ```suggestion
`imagefield` indicates the type of the image you are trying to upload for a particular product. It can be either of these: `front`, `ingredients`, `nutrition`, `packaging` or `other`.
```

```

### Upload Photo of a Product

For authentication, add your valid `user_id` and `password` as body parameters to your request. The `code` (barcode of the product to be updated), `user_id` and `password` are required fields when adding or editing a product. Then, include other product data to be added in the request body.

To write `ingredients_en` to 100% Real Orange Juice so that the image can be uploaded, the request body should contain these fields :

| Key        | Value           | Description  |
| ------------- |:-------------:| -----:|
| user_id     | *** | A valid user_id |
| password      | ***     |   A valid password |
| code | 0180411000803      |    The barcode of the product to be added/edited |
| imagefield | ingredients_en      |    The type of image to be uploaded|
| imageupload_ingredients_en | file     |   The  image of the product ingredients in english |

Using curl:

```bash
curl -XPOST -u off:off -x POST https://world.openfoodfacts.net/cgi/product_jqm2.pl \
  -F user_id=your_user_id -F password=your_password \
  -F code=0180411000803 -F imagefield=ingredients_en -F imageupload_ingredients_en=<binary>"
```

> If the request is succesful, it returns a response that indicated that the fields have been saved.

Reviewer: ```suggestion
If the request is successful, it returns a response that indicated that the fields have been saved.
```

> If the request is successful, it returns a response that indicates that the fields have been saved.

Reviewer 2: ```suggestion
If the request is successful, it returns a response that indicates that the fields have been saved.
You will also get the new image id in `imgid`.
```

```json
{
  "files": [
    {
      "url": "/product/0180411000803/100%-real-orange-juice",
      "filename": "",
      "name": "100% Real Orange Juice",
      "thumbnailUrl": "/images/products/018/041/100/0803.jpg",
      "code": "0180411000803"
    }
  ],
  "image": {
    "thumb_url": "123.100.jpg",
    "imgid": 123,
    "crop_url": "123.400.jpg"
  },
  "imgid": 123,
  "status": "status ok",
  "imagefield": "ingredients_en",
  "code": "0180411000803"
}

> ```

Reviewer 2: ````suggestion
```
If the request is not successful you will have  a dict with an entry `"status": "status not ok"` and an explanation in `debug` field.
````

# Uploading Images to the Open Food Facts API
<!--Add a brief introduction of what the tutorial does -->

> This basic tutorial shows you can upload image of a product to the Open Food Facts API.

Reviewer 3: ```suggestion
This basic tutorial shows you how to upload an image of a product to the Open Food Facts API.
```

## Points to consider before uploading photos

### Image license

> Products images MUST be under the [Creative Commons Attribution ShareAlike licence](https://creativecommons.org/licenses/by-sa/3.0/deed.en)(https://creativecommons.org/licenses/by-sa/3.0/deed.en).

Reviewer 3: ```suggestion
Products images MUST be under the [Creative Commons Attribution ShareAlike licence](https://creativecommons.org/licenses/by-sa/3.0/deed.en).
```

That means you should either upload: 
* photos that are your own work
* photos taken by your users, with their consent for this license (should be part of your service terms)

> * photos already under this license or a more liberal one

Reviewer 3: I would remove the phrasing "or a more liberal one" because it can be easely misinterpreted. Let me know your view on this one though

Author: I am not entirely sure the license context, @alexgarel  will be in the best position to answer.

Reviewer 2: I agree. Let's be more precise to remove ambiguity.

```suggestion
* photos already under this license or a compatible license ([cc-by](https://creativecommons.org/licenses/by/4.0/), [cc-0](https://creativecommons.org/share-your-work/public-domain/cc0/) or public domain)
```

### Image Quality

Uploading quality photos of a product, its ingredients and nutrition table is very important, since it allows the Open Food Facts OCR system to retrieve important data to analyze the product. The minimal allowed size for photos is 640 x 160 px.

### Upload Behavior

> In case you upload more than one photo of the front, the ingredients and the nutrition facts, beware that only the first photo of each category will be displayed. (You might want to take additional images of labels, recycling instructions, and so on). However, all photos will be saved.

Reviewer 3: ```suggestion
In case you upload more than one photo of the front, the ingredients, the nutrition facts, or the product packaging components, beware that only the latest "selected" photo of each category will be displayed on the product page on the website and on the mobile application. 

The older ones are still saved, and can be "selected" either by an API call or via the editing interface (website and mobile application).

You can also upload some photos that are neither of that 4 categories, but they will not be displayed by default.. However, all photos will be saved.
```

### Label Languages

Multilingual products have several photos based on languages present on the packaging. You can specify the language by adding a lang code suffix to the image field.

See [the paragraph about Imagefield](#imagefield)

## Authentication

> The WRITE operations in the OFF API require authentication, therefore you need a valid `user_id` and `password`  to write the photo to 100% Real Orange Juice.

Reviewer 3: ```suggestion
The WRITE operations in the OFF API require authentication, therefore you need a valid `user_id` and `password` to write the photo to 100% Real Orange Juice.
```
